src/main/java/com/example/bistro/bistroapp/console/ConsoleHandlerProduct.java

```diff
@@ -0,0 +1,143 @@
+
+import com.example.bistro.bistroapp.entity.Ingredient;
+import com.example.bistro.bistroapp.entity.Product;
+import com.example.bistro.bistroapp.entity.ProductType;
+import com.example.bistro.bistroapp.service.CustomerService;
+import com.example.bistro.bistroapp.service.IngredientService;
+import com.example.bistro.bistroapp.service.ProductService;
+import org.hibernate.Hibernate;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+
+@Component
+public class ConsoleHandlerProduct {
+    private final ProductService productService;
+    private final CustomerService customerService;
+    private final IngredientService ingredientService;
+    private final Scanner scanner = new Scanner(System.in);
+
+    public ConsoleHandlerProduct(ProductService productService, CustomerService customerService, IngredientService ingredientService) {
+        this.productService = productService;
+        this.customerService = customerService;
+        this.ingredientService = ingredientService;
+    }
+
+    public void addProduct() {
+        System.out.println("Enter product type (CAKE, WAFFLES, CROISSANT, DONUT, PASTA, PIZZA, RISOTTO): ");
+        String productTypeStr = scanner.nextLine();
+        ProductType productType = ProductType.valueOf(productTypeStr);
+
+        if (!isProductTypeAllowed(productType)) {
+            System.out.println("Invalid product type. Please choose from: CAKE, WAFFLES, CROISSANT, DONUT, PASTA, PIZZA, RISOTTO");
+            return;
+        }
+        System.out.println("Enter product name: ");
+        String productName = scanner.nextLine();
+
+        System.out.println("Enter product price: ");
+        double productPrice = scanner.nextDouble();
+        scanner.nextLine();
+
+        Product newProduct = new Product();
+        newProduct.setName(productName);
+        newProduct.setPrice(productPrice);
+        newProduct.setType(productType);
+
+        System.out.println("Enter the number of ingredients for the product: ");
+        int numIngredients = scanner.nextInt();
+        scanner.nextLine();
+
+        Set<Ingredient> ingredients = new HashSet<>();
+        for (int i = 0; i < numIngredients; i++) {
+            System.out.println("Enter ingredient ID: ");
+            Long ingredientId = scanner.nextLong();
+            scanner.nextLine();
+
+            Ingredient ingredient = ingredientService.getIngredientById(ingredientId);
+            if (ingredient != null) {
+                ingredients.add(ingredient);
+            } else {
+                System.out.println("Ingredient with ID " + ingredientId + " not found. Skipping...");
+            }
+        }
+
+        newProduct.setIngredients(ingredients);
+
+        Product addedProduct = productService.addProduct(newProduct);
+        System.out.println("Added product with ID: " + addedProduct.getId());
+    }
+    @Transactional
```

Comment:
I can see that you still have @Transactional here. Why do you need it?

Reply:
Without @transactional the program doesn't output all the products when we try to list them.